### ISSUE TITLE
LipSync groups violating RFC5888

### ISSUE THREAD

**Issue by gloinul** [rtcweb-wg/jsep#162]
I. Section 5.3.1:

   The next step is to generate lip sync groups as defined in [RFC5888],
   Section 7.  For each MediaStream with more than one MediaStreamTrack,
   a group of type "LS" MUST be added that contains the mid values for
   each MediaStreamTrack in that MediaStream.  In some cases this may
   result in adding a mid to a given LS group that was not in that LS
   group in the associated offer.  Although this is not allowed by
   [RFC5888], it is allowed when implementing this specification.

So, do I understand it correctly the violation this paragraph is discussing is the following from Section 9.2 of RFC5888:

   The
   identification-tags contained in this "a=group" line MUST be the same
   as those received in the offer, or a subset of them (zero
   identification-tags is a valid subset).

So when will these violations occur for a LS group in JSEP. I assume due to that offer and answer has different structure of how MediaStreams uses the different m= lines. But, isn't this just another case of when an answerer will have to do an follow up Offer to be able to add all the MediaStreamTracks and their MediaStreams due to different sets on the two peers?

I don't really think this specification should override RFC 5888. The obvious solution is to make a new offer by the answering part who couldn't express its desired configuration in the first round.

**Comment by fluffy** [comment]
I suspect we have two paths here.  A) make a 5888 bis that allows what is currently in draft (and this may turn into bis go group semantics too)   B) just use 5888.   B is slightly less effecent in certian cases than A but it seems like that case will not come up often so I'm not sure it matters much. We should talk about how relevant that case is and if it is not a big deal, B is the fastest path to done

**Comment by juberti** [comment]
Agree that B seems like the easiest path, and we can definitely extend it to A later without too much trouble. We will note the issues that exist and how they could be dealt with by future generations.

The overall fundamental question is do we stay with LS or do something else (e.g. MSID) and for now we are going to stick with LS.

**Comment by juberti** [comment]
We probably need to have some text indicating that LS is what determines what MediaStream a given MediaStreamTrack goes into. Ergo, this reduces the applicability of MSID (which is still used to identify that there is a sending track for a m= section)

**Comment by ekr** [comment]
I think this needs to be done by @fluffy. I am happy to wordsmith afterwards

**Comment by fluffy** [comment]
Have a look at https://github.com/rtcweb-wg/jsep/pull/167

**Comment by alvestrand** [comment]
My suggestion for a simple (???) track assignment algorithm here:

https://mailarchive.ietf.org/arch/msg/rtcweb/WY9mtrR4y9io3u2YI72dIw2D30o

**Comment by hardie** [comment]
Need someone to review PR for grammar and editorial usage. @aboba can you take a look?

**Comment by aboba** [comment]
Took a look. It is hard to parse in places. Do you want me to try to suggest improvements?

**Comment by aboba** [comment]
How about this?

"The next step is to generate session-level lip sync groups as defined in ,
Section 7. All Tracks in the same MediaStream need to be placed in the same
"LS" group, as long as these tracks were intended to be synchronized in the
offer. RFC5888 only allows LS groups in the

Answer to be a subset of what was in the Offer. If there are tracks in the
same MediaStream that the Answerer desires to add to the "LS" group, once
the state moves back to "stable" the answering browser MAY initiate another
offer/answer negotiation, putting all the tracks in the MediaStream into
the "LS" group. Without initiating another O/A exchange, the answering
browser cannot add tracks to the LS group that were not in the Offer.
Rather than adding complexity to address this corner
condition within a single offer/answer exchange, the Working Group
recommends that applications
needing to solve this problem utilize an additional round of signaling,
reversing the offer / answer roles."

**Comment by juberti** [comment]
Blocked on #65

**Comment by juberti** [comment]
I was planning to take a shot at this now that the transceiver stuff has landed.
